probe/endpoint/dns_snooper_linux_amd64.go

```diff
@@ -203,6 +207,20 @@ func (s *DNSSnooper) run() {
 	}
 }
 
+// handleDecodeError logs errors up to the maximum allowed count
+func (s *DNSSnooper) handleDecodingError(err error) {
+	str := err.Error()
+	count := s.decodingErrorCounts[str]
+	count++
+	s.decodingErrorCounts[str] = count
```

Comment:
if the error string is variable, e.g. contain some id, or a timestamp, etc, etc, then this becomes a memory leak.